src/lbaf/Applications/LBAF_app.py

```diff
@@ -94,20 +94,22 @@ def parse_conf_file(self):
             if param_key in self.__allowed_config_keys:
                 self.__dict__[param_key] = param_val
 
-        # Parsing exodus if present
-        if self.configuration.get("exodus") is not None:
+        # Parse whether meshes must be generated
+        if (gm := self.configuration.get("generate_meshes")) is not None:
             self.grid_size = []
-            for key in ("x_procs", "y_procs", "z_procs"):
-                self.grid_size.append(self.configuration.get("exodus").get(key))
-            if math.prod(self.grid_size) != self.n_ranks:
-                raise AssertionError(f"Grid size: {self.grid_size} from exodus is not equal to n_ranks: {self.n_ranks}")
+            for key in ("x_ranks", "y_ranks", "z_ranks"):
+                self.grid_size.append(gm.get(key))
+            if math.prod(self.grid_size) < self.n_ranks:
```

Comment:
shouldn't `math.prod(self.grid_size)` be equal to `self.n_ranks`? What if that would be bigger?

Reply:
It _can_ be bigger. We need to have **at least** enough grid vertices to place all ranks. For example say we have 7 ranks, we can place them on a 2x2x2 grid (or 4x2x1, or even 8x1x1). But we cannot place them on a 3x2x1 grid.

Reply:
Ok, that makes sense.